src/browser/BrowserCodeReader.ts

```diff
@@ -704,11 +704,11 @@ export class BrowserCodeReader {
 
         if (!isChecksumOrFormatError && !isNotFound) {
           // not expected
-          throw e;
+          callbackFn(null, e);
+          return;
         }
 
         // trying again
-        callbackFn(null, e);
```

Comment:
Removing this will make impossible to users detect what was the problem that caused the decode to not happen in this loop. I can see why you changed here, but this callback is necessary. Maybe the `callbackFn` could be called before the `if` block anyways and then the `if` just returns, if matches the condition of course, to break the loop.

Reply:
I moved the callback above the if block as requested.

You should consider to unify the error handling behavior between `decodeAsync` and `decodeContinuously in the near future as `decodeAsync` only reports unexpected error cases and ignores the more common cases.

Reply:
I can't do that because decodeAsync should run only once and decode something, it can fail only one time, that's on reject and it could be rejected with a NotFoundException. The continuous decode, on the other side, can notify via callback the result of each attempt and then it can keep sending how much NotFound it has to.